src/js/utils/ajax.js

```diff
@@ -1,140 +1,220 @@
 define([
     'utils/underscore',
-    'utils/validator',
-    'utils/parser',
-    'utils/trycatch'
-], function(_, validator, parser, trycatch) {
-    var ajax = {};
-
-    /** Loads an XML file into a DOM object * */
-    ajax.ajax = function (xmldocpath, completecallback, errorcallback, donotparse) {
-        var xmlhttp;
-        var isError = false;
-        // Hash tags should be removed from the URL since they can't be loaded in IE
-        if (xmldocpath.indexOf('#') > 0) {
-            xmldocpath = xmldocpath.replace(/#.*$/, '');
+    'utils/parser'
+], function(_, parser) {
+    var exports = {};
+    var noop = function(){};
+    var useDomParser = false;
+
+    var crossdomain = exports.crossdomain = function(uri) {
+        var a = document.createElement('a');
+        var b = document.createElement('a');
+        a.href = location.href;
+        try {
+            b.href = uri;
+            b.href = b.href; /* IE fix for relative urls */
+            return a.protocol + '//' + a.host !== b.protocol + '//' + b.host;
+        } catch(e) {}
+        return true;
+    };
+
+    var ajax = exports.ajax = function (url, completeCallback, errorCallback, args) {
+        if (_.isObject(url)) {
```

Comment:
We may want to get rid of this legacy fallback in the future, once we update vast.
Add a comment for it?

    // Legacy support for vast plugin, remove at later point